pkg/generated-assets/generated_assets.go

```diff
@@ -275,3 +275,22 @@ func getYAMLKind(yaml []byte) (string, error) {
 	}
 	return gvk.GroupKind().String(), nil
 }
+
+func sortAssets(names []string) []string {
+	var stage1, stage2, stage3 []string
+	for i := range names {
+		switch {
+		case strings.HasSuffix(names[i], "_sa.yaml") || strings.HasSuffix(names[i], "_role.yaml"):
+			stage1 = append(stage1, names[i])
+		case strings.HasSuffix(names[i], "_binding.yaml"):
```

Comment:
I don't like depending on specific file names. It would be more robust to sort by the yaml content. `GetControllerStaticAssetNames` already parses the yaml files and has their `kind`, it should be possible to sort by it. Something like:

```
func getCreationOrder(kind string) int {
    if rbacKinds.Has(kind) {
        return  0
    }
    return 1
}
```